src/VisualStudio/Core/Def/ValueTracking/ValueTrackingCommandHandler.cs

```diff
@@ -86,101 +85,59 @@ public bool ExecuteCommand(ValueTrackingEditorCommandArgs args, CommandExecution
+            var rootViewModels = await rootItemMap.SelectAsArrayAsync(async (pair, cancellationToken) =>
+            {
+                var (parent, children) = pair;
+
+                if (parent is null)
+                {
+                    // If the parent items are null then each child is a top level item
+                    return await children.SelectAsArrayAsync(async (child, cancellationToken) =>
+                    {
+                        return await ValueTrackedTreeItemViewModel.CreateAsync(solution, child, toolWindow.ViewModel, _glyphService, valueTrackingService, _threadingContext, ImmutableArray<TreeItemViewModel>.Empty, cancellationToken).ConfigureAwait(false);
+                    }, cancellationToken).ConfigureAwait(false);
+                }
 
-            var childViewModels = await childItems.SelectAsArrayAsync((item, cancellationToken) =>
+                var childrenViewModels = await children.SelectAsArrayAsync((item, cancellationToken) =>
                 ValueTrackedTreeItemViewModel.CreateAsync(solution, item, toolWindow.ViewModel, _glyphService, valueTrackingService, _threadingContext, cancellationToken), cancellationToken).ConfigureAwait(false);
 
-            RoslynDebug.AssertNotNull(location.SourceTree);
-            var document = solution.GetRequiredDocument(location.SourceTree);
-            var options = ClassificationOptions.From(document.Project);
-
-            var sourceText = await location.SourceTree.GetTextAsync(cancellationToken).ConfigureAwait(false);
-            var documentSpan = await ClassifiedSpansAndHighlightSpanFactory.GetClassifiedDocumentSpanAsync(document, location.SourceSpan, options, cancellationToken).ConfigureAwait(false);
-            var classificationResult = await ClassifiedSpansAndHighlightSpanFactory.ClassifyAsync(documentSpan, options, cancellationToken).ConfigureAwait(false);
-
-            var root = new TreeItemViewModel(
-                location.SourceSpan,
-                sourceText,
-                document.Id,
-                document.FilePath ?? document.Name,
-                selectedSymbol.GetGlyph(),
-                classificationResult.ClassifiedSpans,
-                toolWindow.ViewModel,
-                _glyphService,
-                _threadingContext,
-                solution.Workspace,
-                childViewModels);
+                var root = await ValueTrackedTreeItemViewModel.CreateAsync(solution, parent, toolWindow.ViewModel, _glyphService, valueTrackingService, _threadingContext, childrenViewModels, cancellationToken).ConfigureAwait(false);
+                return ImmutableArray.Create(root);
+            }, cancellationToken).ConfigureAwait(false);
 
             await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
             toolWindow.ViewModel.Roots.Clear();
-            toolWindow.ViewModel.Roots.Add(root);
+            foreach (var root in rootViewModels.SelectMany(items => items))
```

Comment:
>SelectMany [](http://example.com/codeflow?start=48&length=10)

Seems like it would be simpler to write the above code as a foreach that adds items to an ArrayBuilder since this is flattening them out anyways.